src/query/storages/common/index/src/bloom.rs

```diff
@@ -125,42 +137,100 @@ impl BlockFilter {
             return Err(ErrorCode::BadArguments("block is empty"));
         }
 
+        let mut exprs = Vec::new();
+        let mut fields = Vec::new();
+        let mut columns = Vec::new();
+        for i in 0..blocks[0].num_columns() {
+            let data_type = &blocks[0].get_by_offset(i).data_type;
+            if Xor8Filter::is_supported_type(data_type) {
+                let source_field = source_schema.field(i);
+                let return_type = if data_type.is_nullable() {
+                    DataType::Nullable(Box::new(DataType::Number(NumberDataType::UInt64)))
+                } else {
+                    DataType::Number(NumberDataType::UInt64)
+                };
+                let field = DataField::new(source_field.name().as_str(), return_type.clone());
+                fields.push(field);
+
+                let arg = PhysicalScalar::IndexedVariable {
+                    index: columns.len(),
+                    data_type: data_type.clone(),
+                    display_name: source_field.name().clone(),
+                };
+                let scalar = PhysicalScalar::Function {
+                    name: "siphash".to_string(),
+                    params: vec![],
+                    args: vec![arg],
+                    return_type,
+                };
+                let expr = scalar.as_expr()?;
+                exprs.push(expr);
+
+                let source_columns = blocks
```

Comment:
`concat` already cover the convert full column codes.

Reply:
use `DataBlock::concat` ?